control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -581,6 +593,26 @@ func (r *GatewayController) transformConsulHTTPRoute(ctx context.Context) func(e
 	}
 }
 
+// transformGatewayPolicy will return a list of all gateways that need to be reconcilled.
+func (r *GatewayController) transformGatewayPolicy(ctx context.Context) func(object client.Object) []reconcile.Request {
+	return func(o client.Object) []reconcile.Request {
+		gatewayPolicy := o.(*v1alpha1.GatewayPolicy)
+
+		//gateway policy to gateway is 1to1
```

Comment:
this isn't necessarily the case, it's `gatewayPolicy:gatewayListener` is 1:1, so multiple policies can reference a gateway at this point, but each listener can only be referenced once